GVFS/GVFS.Common/Git/CredentialManager.cs

```diff
@@ -0,0 +1,29 @@
+﻿using GVFS.Common.Tracing;
+
+namespace GVFS.Common.Git
+{
+    public class CredentialManager : ICredentialManager

```

Comment:
Minor: Unless we plan on having multiple `CredentialManager` implementations in the product, can you remove the `ICredentialManager` interface and make the methods you want to override for unit testing virtual?  That's the approach we're using elsewhere.

Reply:
We have only one implementation of `ITracer` which is `JsonTracer`?
I'm all for the interface in my vote.

Reply:
> We have only one implementation of ITracer which is JsonTracer?
> I'm all for the interface in my vote.

You're right about `ITracer/JsonTracer`.  It predates the guideline to avoid interfaces whose only purpose is unit testing and no one went back to make that class/interface consistent.  

The other interfaces in the code are either for cross-platform support, or are used when we need to expose a small surface area of an underlying class (e.g. `IProfilerOnlyIndexProjection`, `IHeartBeatMetadataProvider`).  

I think that @kewillford's suggestion for `ICredentialManager` is a better fit for the style used elsewhere in VFS4G:  

> You could have the GitProcess be an ICredentialManager and remove this class all together. I would only have this class if it was doing something besides being a pass through.

The [current guideline](https://github.com/Microsoft/VFSForGit/blob/master/CONTRIBUTING.md) is to avoid adding interfaces whose only purpose is unit testing.  If there's agreement the guideline should change then there should be a separate PR to update the guideline (and source code to match the new guideline).

Reply:
> I think that @kewillford's suggestion for ICredentialManager is a better fit for the style used elsewhere in VFS4G

That is the approach taken in now.